src/stdlib_logger.f90

```diff
@@ -758,46 +813,29 @@ subroutine log_error( self, message, module, procedure, stat, errmsg )
         integer :: unit
         integer :: iostat
         character(*), parameter :: procedure_name = 'log_error'
-        character(256) :: iomsg
+        character(256) :: iomsg, suffix
 
-        call self % log_message( message,               &
-                                 module = module,       &
-                                 procedure = procedure, &
-                                 prefix = 'ERROR')
-
-        if ( self % units == 0 ) then
-            call write_log_error( output_unit )
-        else
-            do unit=1, self % units
-                call write_log_error( self % log_units(unit) )
-            end do
+        if ( present(stat) ) then
+            write( suffix, '(a, i0)', err=999, iostat=iostat, iomsg=iomsg ) &
+                new_line('a') // "With stat = ", stat
         end if
 
-    contains
-
-        subroutine write_log_error( unit )
-            integer, intent(in) :: unit
-
-            if ( present(stat) ) then
-                write( unit, '("With stat = ", i0)', err=999, &
-                    iostat=iostat, iomsg=iomsg ) stat
-            end if
-
-            if ( present(errmsg) ) then
-                if ( len_trim(errmsg) > 0 ) then
-                    call format_output_string( self, unit,          &
-                                               'With errmsg = "' // &
-                                               trim(errmsg) // '"', &
-                                               procedure_name,      &
-                                               '    ' )
-                end if
+        if ( present(errmsg) ) then
+            if ( len_trim(errmsg) > 0 ) then
+                suffix( len_trim(suffix)+1: ) = &
+                    new_line('a') // 'With errmsg = "' // trim(errmsg) // '"'
```

Comment:
If the length of `errmsg` is > 256-15, then it cannot be contained in `suffix`. Would it not be preferable to make `suffix` allocatable such that it would be possible to adapt its length depending on the length of `errmsg`?

Reply:
I guess what we could do is define two character variables, say suffix and dummy, with suffix an allocatable len variable. We could size dummy large enough to hold the output up to this point, and allocate suffix with the string assignment `trim(dummy) // new_line('a') // 'With errmsg = "' // trim(errmsg) // '"'`.  If you want I could implement this while you take a break.

Reply:
This will have to be done in two locations: `log_error` and `log_io_error`.